File: python/Ball.py
from CaptureDevice import CaptureDevice

# buat kelas ball
class Ball(CaptureDevice):
    def __init__(self, id_Item: str = "", nama_Item: str = "", harga_item: float = 0.0, deskripsi_Item: str = "",
                 levelCap: str = "", typeDevice: str = "",
                 kondisiBall: str = "", multiplierBall: float = 0.0, pastiDapetBall: int = 0):
        super().__init__(id_Item, nama_Item, harga_item, deskripsi_Item, levelCap, typeDevice)
        self.__kondisi_Ball = str(kondisiBall)            # buat nambah kondisi ball
        self.__multiplier_Ball = float(multiplierBall)    # buat nambah multiplier ball
        self.__pastiDapet_Ball = int(pastiDapetBall)      # buat nambah pasti dapet ball

    # buat setter kondisi ball
    def setKondisi_Ball(self, kondisiBall):
        self.__kondisi_Ball = str(kondisiBall)

    def setKondisiBall(self, kondisiBall):
        self.__kondisi_Ball = str(kondisiBall)

    # buat setter multiplier ball
    def setMultiplier_Ball(self, multiplierBall):
        self.__multiplier_Ball = float(multiplierBall)

    def setMultiplierBall(self, multiplierBall):
        self.__multiplier_Ball = float(multiplierBall)

    # buat setter pasti dapet ball
    def setPastiDapet_Ball(self, pastiDapetBall):
        self.__pastiDapet_Ball = int(pastiDapetBall)

    def setPastiDapetBall(self, pastiDapetBall):
        self.__pastiDapet_Ball = int(pastiDapetBall)

    # buat getter kondisi ball
    def getKondisi_Ball(self):
        return self.__kondisi_Ball

    def getKondisiBall(self):
        return self.__kondisi_Ball

    # buat getter multiplier ball
    def getMultiplier_Ball(self):
        return self.__multiplier_Ball

    def getMultiplierBall(self):
        return self.__multiplier_Ball

    # buat getter pasti dapet ball
    def getPastiDapet_Ball(self):
        return self.__pastiDapet_Ball

    def getPastiDapetBall(self):
        return self.__pastiDapet_Ball

    # buat cek status pasti dapat
    def getKeteranganPastiDapet(self):
        if self.__pastiDapet_Ball == 1:
            return "Pasti dapat Pokemon"
        else:
            return "Tidak pasti dapat Pokemon"

    # buat cek pasti dapat
    def cekPastiDapat(self):
        return self.__pastiDapet_Ball == 1

    # buat show data ball
    def showDataBall(self):
        print(f"Kondisi Ball       : {self.__kondisi_Ball}")
        print(f"Multiplier Ball    : {self.__multiplier_Ball:.2f}x")
        pasti_str = "1 (Pasti dapat Pokemon)" if self.__pastiDapet_Ball == 1 else "0 (Tidak pasti dapat Pokemon)"
        print(f"Pasti Dapet        : {pasti_str}")

    # buat show data lengkap
    def showData(self):
        print("----------------------------------------")
        print("           DATA POKEBALL ITEM           ")
        print("----------------------------------------")
        super().showData()
        self.showDataCaptureDevice()
        self.showDataBall()
        print("----------------------------------------")

File: php/Ball.php
<?php
require_once 'CaptureDevice.php';

// buat kelas ball
class Ball extends CaptureDevice {
    private string $kondisi_Ball;    // buat nambah kondisi ball
    private float $multiplier_Ball;  // buat nambah multiplier ball
    private int $pastiDapet_Ball;    // buat nambah pasti dapet ball

    // constructor default & parametrik
    public function __construct(
        string $id = "",
        string $nama = "",
        float $harga = 0.0,
        string $deskripsi = "",
        string $gambar = "",
        string $levelCap = "",
        string $type = "",
        int $isUsable = 0,
        string $kondisi = "",
        float $multiplier = 0.0,
        int $pastiDapet = 0
    ) {
        parent::__construct($id, $nama, $harga, $deskripsi, $gambar, $levelCap, $type, $isUsable);
        $this->kondisi_Ball = $kondisi;
        $this->multiplier_Ball = $multiplier;
        $this->pastiDapet_Ball = $pastiDapet;
    }

    // buat setter kondisi ball
    public function setKondisi_Ball(string $kondisi): void {
        $this->kondisi_Ball = $kondisi;
    }

    public function setKondisiBall(string $kondisi): void {
        $this->kondisi_Ball = $kondisi;
    }

    // buat setter multiplier ball
    public function setMultiplier_Ball(float $multiplier): void {
        $this->multiplier_Ball = $multiplier;
    }

    public function setMultiplierBall(float $multiplier): void {
        $this->multiplier_Ball = $multiplier;
    }

    // buat setter pasti dapet ball
    public function setPastiDapet_Ball(int $pastiDapet): void {
        $this->pastiDapet_Ball = $pastiDapet;
    }

    public function setPastiDapetBall(int $pastiDapet): void {
        $this->pastiDapet_Ball = $pastiDapet;
    }

    // buat getter kondisi ball
    public function getKondisi_Ball(): string {
        return $this->kondisi_Ball;
    }

    public function getKondisiBall(): string {
        return $this->kondisi_Ball;
    }

    // buat getter multiplier ball
    public function getMultiplier_Ball(): float {
        return $this->multiplier_Ball;
    }

    public function getMultiplierBall(): float {
        return $this->multiplier_Ball;
    }

    // buat getter pasti dapet ball
    public function getPastiDapet_Ball(): int {
        return $this->pastiDapet_Ball;
    }

    public function getPastiDapetBall(): int {
        return $this->pastiDapet_Ball;
    }

    // buat cek status pasti dapat
    public function getKeteranganPastiDapet(): string {
        if ($this->pastiDapet_Ball === 1) {
            return "Pasti dapat Pokemon";
        } else {
            return "Tidak pasti dapat Pokemon";
        }
    }

    // buat cek pasti dapat
    public function cekPastiDapat(): bool {
        return $this->pastiDapet_Ball === 1;
    }

    // buat show data ball
    public function showDataBall(): void {
        echo "Kondisi Ball       : " . $this->kondisi_Ball . "<br>";
        echo "Multiplier Ball    : " . number_format($this->multiplier_Ball, 2) . "x<br>";
        $pastiStr = ($this->pastiDapet_Ball === 1) ? "1 (Pasti dapat Pokemon)" : "0 (Tidak pasti dapat Pokemon)";
        echo "Pasti Dapet        : " . $pastiStr . "<br>";
    }

    // buat show data lengkap
    public function showData(): void {
        echo "----------------------------------------<br>";
        echo "           DATA POKEBALL ITEM           <br>";
        echo "----------------------------------------<br>";
        parent::showData();
        $this->showDataCaptureDevice();
        $this->showDataBall();
        echo "----------------------------------------<br>";
    }
}
?>
